tests/unit/test_endpoint_aggregator.py

```diff
@@ -78,11 +77,10 @@ def __init__(self, *args, **kwargs):
         )
 
 
-@patch("charms.observability_libs.v0.juju_topology.JujuTopology.is_valid_uuid", lambda *args: True)
 class TestEndpointAggregator(unittest.TestCase):
     def setUp(self):
         self.harness = Harness(EndpointAggregatorCharm, meta=AGGREGATOR_META)
-        self.harness.set_model_info(name="testmodel", uuid="1234567")
+        self.harness.set_model_info(name="testmodel", uuid="12de4fae-06cc-4ceb-9089-567be09fec78")
```

Comment:
I like this PR Maksim!
I was thinking... what about generating the UUID every time we run the test? Seems it will be more robust.

Reply:
then we will need to regenerate model names as well, since they contain part of the UUID.

eg
https://github.com/canonical/prometheus-k8s-operator/blob/07074183ec918d9a8db10f09b1bb7133d7140eff/tests/unit/test_endpoint_aggregator.py#L114

I think logic is clear, we can just trim UUID to the first 7 chars, if I got it right

but before going there, I would like to see what other think

@rbarry82 , @simskij @dstathis

Reply:
I tend to agree that just trimming is probably enough. That's what `observability_libs.JujuTopology` does anyway, and the added complexity in dynamic creation of the rest of the attributes (model name, etc) isn't worth it from my POV considering that the unit tests will never _truly_ be able to robustly make assertions against Juju anyway. That requires integration tests.

From my side, I'm ok with unit tests which demonstrate that the essential logical flows work with reasonable values, even if they're static/const.